rcl_logging_spdlog/test/test_logging_interface.cpp

```diff
@@ -62,6 +65,42 @@ class RestoreEnvVar
   const std::string value_;
 };
 
+class TemporaryLogDir final
+{
+public:
+  TemporaryLogDir()
+  : orig_ros_log_dir_value_(rcpputils::get_env_var("ROS_LOG_DIR"))
+  {
+    // Create a random directory in our current path
+    std::random_device dev;
+    std::mt19937 prng(dev());
+    std::uniform_int_distribution<uint64_t> rand(0);
+
+    std::stringstream ss;
+    ss << std::hex << rand(prng);
+
+    local_log_dir_ = std::filesystem::current_path() / ss.str();
+
+    if (!std::filesystem::create_directories(local_log_dir_)) {
+      throw std::runtime_error("Failed to make temporary log directory");
+    }
+
+    rcpputils::set_env_var("ROS_LOG_DIR", local_log_dir_.string().c_str());
+  }
+
+  ~TemporaryLogDir()
+  {
+    rcpputils::set_env_var("ROS_LOG_DIR", orig_ros_log_dir_value_.c_str());
+    if (std::filesystem::remove_all(local_log_dir_) == 0) {
+      std::cerr << "Failed to remove temporary directory\n";
+    }
```

Comment:
> We may end up with a few extra temporary directories, but wouldn't it be easier to just integrate this directly into the `LoggingTest` fixture?

I could do that; I'll take a look.  I did it this way since we don't *have* to do this in all of the tests, but it probably doesn't hurt.

> I'm nervous about doing this sort of stuff in a C++ destructor.

But I think this is what is going to happen anyway.  When the fixture gets destroyed, it will do the `remove_all` at that point.

Reply:
> I tried to do this, but the problem is that part of these tests is to use an invalid allocator. And when we do that, we have a separate bug down in rcutils_strdup where it doesn't handle a zero-initialized allocator properly. We don't have that problem with the separate class because we only use the separate class when we aren't using the invalid allocator.

This is a bug in `rcl_logging_spdlog` too, and honestly we're just getting lucky right now. Simply setting `ROS_HOME` to anything prior to invoking this test will cause it to segfault. I opened a [PR to address this](https://github.com/ros2/rcl_logging/pull/116).



> > I'm nervous about doing this sort of stuff in a C++ destructor.
>
> But I think this is what is going to happen anyway. When the fixture gets destroyed, it will do the remove_all at that point.

My concern isn't about whether or not to do it, it's about specifically doing it in a C++ class destructor. At the very least, this implementation lets exceptions escape the destructor. It just feels like this sort of activity belongs in a test fixture - it's what they're for. Even making a simple LoggingTestWithTemp fixture that inherits from LoggingTest and adds the temporary directory functionality there would be much cleaner and wouldn't add any more code than this.

Reply:
OK, that's fair.  I actually ended up simplifying this a bit in a different way.  We already have two fixtures, and we really only need the AllocatorTest fixture when we are testing bad fixtures.  So I split the tests along those lines, and then just made `LoggingTest` unconditionally create and destroy the temporary directory.

When you get a chance, take another look.